### PR TITLE
chore: keda abstraction layer on `be-web` chart

### PR DESCRIPTION
**Pull request by gi0baro**
SSIA

### CODE REVIEW

**Review by giuscri — COMMENTED** [review]
@edubacco What do you think?

File: charts/be-web/templates/scaledobject.yaml [hunk 1]
```diff
@@ -2,11 +2,46 @@
 apiVersion: keda.sh/v1alpha1
 kind: ScaledObject
 metadata:
-  name: {{ template "app.fullname" . }}
+  name: {{ include "app.fullname" . }}
   labels:
     {{- include "app.labels" . | nindent 4 }}
 spec:
-  {{- with .Values.autoscaling.keda.spec }}
+  scaleTargetRef:
```
Comment (edubacco) [inline charts/be-web/templates/scaledobject.yaml:9]: we should add `cooldownPeriod`

**Review by edubacco — APPROVED** [review]
Generally speaking, I continue to think that the approach is too complex.

To use keda with the chart, the dev should:
1 - read keda documentation and understand his options
2 - read the chart templates, and understand which option is supported and how
3a - open a PR to support his use case
3 - implement values.yaml

I do not understand which are the benefits of the point 2, and consequently 3a

With no abstraction, the flow (and the cognitive load on the developer) will be easier:
1 - read keda documentation and understand his options
2 - implement values.yaml

Let's keep in mind that we are speaking about something new, where use cases and possibilities are various and unknown. The advantages of an abstraction IMHO are that you can push implementations to known use cases, and limit possibilities to avoid problems. In this case, I think that limiting options/usecases is not desiderable, it's a premature optimization, and so the whole abstraction seems a style exercise rather than something useful.

Said that, this implementation works for my specific need, and as we need to converge to _a_ solution, I will approve the pr to be able to go on.

**Comment by gi0baro** [comment]
> I do not understand which are the benefits of the point 2, and consequently 3a

@edubacco IMHO point 2 and 3a won't never happen.
The _moving part_ of Keda configuration is related to triggers, not really in Scaling objects.
And that's the point of the abstraction layer: you just need to document yourself on the trigger, not the custom resources in Kubernetes produced by Keda.

In my experience – of course you need to trust me on this, but on the other side: why should I lie? – application developers need to customise the scaled object properties outside of the triggers less than 1% of an application lifetime.
Obeying developers to always know the entire spec of the scaled objects IMHO is not a better solution, as you might end-up with copy-pasting rather than documenting.

Aside from this, I gonna add polling interval and cooldown options and merge.

**Pull request merged**